base/path.jl

```diff
@@ -259,20 +321,7 @@ julia> joinpath("/home/myuser", "example.jl")
 "/home/myuser/example.jl"
 ```
 """
-joinpath(a::AbstractString, b::AbstractString, c::AbstractString...) = joinpath(joinpath(a,b), c...)
-joinpath(a::AbstractString) = a
-
-function joinpath(a::String, b::String)
```

Comment:
This code is very kludgy, considering how straightforward the code is on non-Windows systems, which is  especially true since we need  a custom Windows implementation anyways.

The improved code is about 20x faster in common cases, and allocates far fewer.

Reply:
Calling the code that someone else wrote kludgy when it is very likely that person who you'll need to review your code seems like a bad way to approach getting changes reviewed. There's no need to shit on the existing code in order to make improvements.

Reply:
I deeply apologize to anyone if this came across offensive, as it was definitely not  intention to do so.   I just found the previous logic  hard to follow and confusing, which is what I mean when I said "kludgy"; I totally didn't mean anything malicious (poor comment on my behalf).